org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/semanticTokens/SemanticHighlightReconcilerStrategyTest.java

```diff
@@ -61,11 +61,11 @@ public void testKeyword() throws CoreException {
 		DisplayHelper.sleep(display, 2_000); // Give some time to the editor to update
 
 		StyleRange[] styleRanges = textViewer.getTextWidget().getStyleRanges();
-
+		//default theme is dark, that's why we have to expect a blue foreground color:
```

Comment:
I don't think this is generally true. The default theme may depend on other OS settings. We need something more flexible (eg evaluating whether theme is dark or not), or more constraining (eg setting the theme) here as we cannot reliably assume what the default theme is.

Reply:
Agree. Wouldn't it be enough to check the start and length of the style range?

Reply:
Yes, checking ranges + `color != widget.getBackgroundColor()` would be good enough